test/replicaset-luatest/replicaset_3_test.lua

```diff
@@ -2,6 +2,7 @@ local fiber = require('fiber')
 local t = require('luatest')
```

Comment:
> Let's validate name only during connection establishing, not on every reconnect

Please, do it on every reconnect. It is not as expensive as it looks, but can protect from accidental URL changes with nasty consequences when the underlying instance could change to a different one if somebody messed up DNS.

Reply:
Now it's validated on every reconnect